Include profile pic/remove username in connections page

darrenvong opened the pull request:
Literally swapping what's on the friends list to re-use the existing `UserSummary` component for this. Achieves the "remove username" and "show profile photo" in one. Since there aren't any designs for this, I thought this is the easiest way to do this and also for maintaining visual consistency. Thoughts?

Closes #761

**Web**
![Screenshot 2021-03-16 at 00 34 04](https://user-images.githubusercontent.com/13669362/111238682-744fd900-85ef-11eb-8929-5694c42cc518.png)

**Mobile Web**
![Screenshot 2021-03-16 at 00 34 29](https://user-images.githubusercontent.com/13669362/111238707-80d43180-85ef-11eb-9f0a-4a551fe84b88.png)

darrenvong committed:
feat: Include profile pic/remove username in connections page

darrenvong committed:
feat: accessibility fix on tab bar's broken aria label

andreaowu commented:
Looks good, can you include a screenshot?

lucaslcode commented on app/frontend/src/components/TabBar/TabBar.tsx:
```diff
@@ -15,7 +18,8 @@ export default function TabBar<T extends Record<string, React.ReactNode>>({
   };
   return (
-    <Tabs
+    <TabList
```
What is the difference?

darrenvong replied:
This injects/handles the right aria props for accessibility according to the docs - https://material-ui.com/components/tabs/#experimental-api - so I thought why not use it. Otherwise I think it's the same as I believe `TabList` uses `Tabs`

lucaslcode reviewed (commented):
The top text of `UserSummary` is H2 which isn't really right on this page...

darrenvong commented:
> Looks good, can you include a screenshot?

@andreaowu Updated original message with screenshots. I need to be better at doing this...

> The top text of `UserSummary` is H2 which isn't really right on this page...

Hmm good point. At least it's not skipping levels but doesn't exactly feel right either. I think I'll alter it to allow the heading level to be overriden from props, since it's impossible for the component on its own to know where it'll be used, which determines what heading level it should be at...

darrenvong committed:
feat: fix missing tab context in messages

darrenvong committed:
feat: Allow element type of user headline to be changed in UserSummary

This is because there's no way for a standalone component like this to
know the correct heading level to use without the context it's used in.
Consuming components have this context and so should provide it to
override the default h2 where appropriate.
